tests/unit/syslog_template_test.py

```diff
@@ -0,0 +1,1037 @@
+                    "proto": "udp",
+                    "srcip": "127.0.0.1",
+                    "destport": self.__class__.udp_port,
+                    "hostname": "localhost",
+                    "appname": "demo",
+                },
+                "parser": "syslog-parser",
+            },
+        )
+
+    def test_proto_destport_param_two_logconfigs(self):
+        self.create_monitor(
+            {
+                "module": "scalyr_agent.builtin_monitors.syslog_monitor",
+                "protocols": "tcp:%d, tcp:%d"
+                % (self.__class__.tcp_port, self.__class__.tcp_port + 1),
+                "message_log_template": "syslog-${PROTO}-${DESTPORT}.log",
+            },
+            [
+                {
+                    "path": os.path.join(
+                        ConfigurationMock.log_path,
+                        "syslog-tcp-%d.log" % self.__class__.tcp_port,
+                    ),
+                    "attributes": JsonObject({"parser": "first-parser"}),
+                },
+                {
+                    "path": os.path.join(
+                        ConfigurationMock.log_path,
+                        "syslog-tcp-%d.log" % (self.__class__.tcp_port + 1,),
+                    ),
+                    "attributes": JsonObject({"parser": "second-parser"}),
+                },
+            ],
+        )
+
+        self.connect_and_send(
+            b"<1>Jan 02 12:34:56 localhost demo[1]: hello world\n",
+            self.__class__.tcp_port,
+        )
+        self.monitor.wait_until_count(1)
+        self.connect_and_send(
+            b"<1>Jan 02 12:34:56 localhost demo[1]: hello world\n",
+            self.__class__.tcp_port + 1,
+        )
+        self.monitor.wait_until_count(2)
+
+        self.assertEqual(len(self.watcher.log_configs), 2)
+        self.assertDictEqual(
+            self.watcher.log_configs[0],
+            {
+                "path": "./syslog-tcp-%d.log" % self.__class__.tcp_port,
+                "attributes": {
+                    "proto": "tcp",
+                    "srcip": "127.0.0.1",
+                    "destport": self.__class__.tcp_port,
+                    "hostname": "localhost",
+                    "appname": "demo",
+                },
+                "parser": "first-parser",
+            },
+        )
+        self.assertDictEqual(
+            self.watcher.log_configs[1],
+            {
+                "path": "./syslog-tcp-%d.log" % (self.__class__.tcp_port + 1,),
+                "attributes": {
+                    "proto": "tcp",
+                    "srcip": "127.0.0.1",
+                    "destport": self.__class__.tcp_port + 1,
+                    "hostname": "localhost",
+                    "appname": "demo",
+                },
+                "parser": "second-parser",
+            },
+        )
+
+    # On Debian based distros `localhost 127.0.1.1` is added to /etc/hosts
+    @skip("Requires aliasing the loopback interface")
```

Comment:
Should we do this skip conditional to make it work on Linux?

Reply:
We could, I'm not sure if non-Debian distros do the same thing or not.

I think I'll just leave it as plain skip for now & if there are problems with that feature in the future will work something out to ensure it's tested.